dangerzone/logic.py

```diff
@@ -36,3 +38,35 @@ def __init__(self) -> None:
 
         # Load settings
         self.settings = Settings(self)
+
+        self.documents: List[Document] = []
+
+    def add_document(
+        self, input_filename: str, output_filename: Optional[str] = None
+    ) -> None:
+        doc = Document(input_filename)
+        if output_filename:
+            doc.output_filename = output_filename
+        else:
+            doc.set_default_output_filename()
+        self.documents.append(doc)
+
+    def convert_documents(
+        self, ocr_lang: Optional[str], stdout_callback: Optional[Callable] = None
+    ) -> None:
+        def convert_doc(document: Document) -> None:
+            success = container.convert(
+                document,
+                ocr_lang,
+                stdout_callback,
+            )
+
+        max_jobs = container.get_max_parallel_conversions()
```

Comment:
> I don't think we need to expose it at this point in time. I'd prefer a UX review first over which functionalities should be exposed to the user. Otherwise we may end up [like pied piper](https://featureisnotabug.files.wordpress.com/2018/02/interface_large.jpg?w=840). 

Right. We would need some sort of "advanced tab", if we wanted to implement this.

> The OS already should do this kind of management anyways, no?

I think that VMs have higher CPU priority than common processes, which is why the OS is not so great at scheduling those.

> Or maybe better is contextual help. Implement a prompt or some sort of warning that the conversion is consuming too many resources and give the user a prompt to reduce the number of processes with a simple click of a button.

It will be difficult to find an adequate heuristic for that. There are cases where using 100% of the CPU is acceptable, and others when it's not. It's similar to the CPU knob of Docker Desktop. Instinctively, most would prefer assigning all the CPU cores to it, but there are cases where this is not desired.

Reply:
>  But that's for another PR.

Ok, I agree. Worst case scenario, users can limit the CPUs on the Docker Desktop VM, if system performance is suffering.